odoo/tools/lru.py

```diff
@@ -31,29 +40,66 @@ def get(self, obj, val=None):
 
     @locked
     def __getitem__(self, obj):
-        a = self.d[obj]
+        a = self.d[obj][0]
         self.d.move_to_end(obj, last=False)
         return a
 
     @locked
     def __setitem__(self, obj, val):
-        self.d[obj] = val
+        footprint = get_memory_footprint(val) + get_memory_footprint(obj) if self.mode == 'footprint' else None
+        self.d[obj] = (val, footprint)
```

Comment:
An alternative solution to avoid the pair is to store the footprint in its own dict (only for mode `footprint`).  I don't know which one would be better, a dict of pairs or a pair of dicts ?